### PR TITLE
feat: implement --format and `SET FORMAT <pretty,csv,json>` in SQL repl

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
Make it easier to consume the output of `influxdb_iox sql` in scripts.

Realistically, we probably also need a "quiet" mode that doesn't print out status (e.g. don't print out `Connected to IOx Server at http://localhost:1234` etc) but this is a step towards it

# Changes
1. Add a `SET FORMAT` command for interactive use
2. Add a `--format` command line argument for scripting use

# Example Use

```
./target/debug/influxdb_iox sql --format=csv --host=http://localhost:1234

...

foobar_weather> show tables;
show tables;
table_catalog,table_schema,table_name,table_type
public,system,chunks,BASE TABLE
public,system,columns,BASE TABLE
public,system,operations,BASE TABLE
public,information_schema,tables,VIEW
public,information_schema,columns,VIEW

Query execution complete in 46.036373ms

foobar_weather> set format pretty;
set format pretty;
Set output format format to pretty
foobar_weather> show tables;
show tables;
+---------------+--------------------+------------+------------+
| table_catalog | table_schema       | table_name | table_type |
+---------------+--------------------+------------+------------+
| public        | system             | chunks     | BASE TABLE |
| public        | system             | columns    | BASE TABLE |
| public        | system             | operations | BASE TABLE |
| public        | information_schema | tables     | VIEW       |
| public        | information_schema | columns    | VIEW       |
+---------------+--------------------+------------+------------+

Query execution complete in 1.382642637s
```

### CODE REVIEW

**Comment by crepererum** [comment]
```text
foobar_weather> show tables;
show tables;
+---------------+--------------------+------------+------------+
| table_catalog | table_schema       | table_name | table_type |
+---------------+--------------------+------------+------------+
| public        | system             | chunks     | BASE TABLE |
| public        | system             | columns    | BASE TABLE |
| public        | system             | operations | BASE TABLE |
| public        | information_schema | tables     | VIEW       |
| public        | information_schema | columns    | VIEW       |
+---------------+--------------------+------------+------------+
```

:top: That's wrong, isn't it?

Also: it would probably be helpful to add some kind of start and end markers before and after the CSV if it is used during some interactive session (like `--- CSV START ---`/`--- CSV END ---`).

**Comment by alamb** [comment]
> 🔝 That's wrong, isn't it?

@crepererum  can you be more specific of what you think is wrong?

**Comment by crepererum** [comment]
Sorry, should have been more precise: why does `show tables` not emit a CSV but a pretty-printed table?

**Comment by alamb** [comment]
> Sorry, should have been more precise: why does show tables not emit a CSV but a pretty-printed table?

Ah, sorry -- I was probably being too clever. The example is meant to show the output of `show tables` both 
1. in csv format (due to starting with the `--format=csv` flag)
2. in pretty format (due to running `set format pretty` from the command line)

Here is annotated output to try and make that clearer:


```
...

foobar_weather> show tables;       <-------------------- Output of initial show tables is in csv format
show tables;
table_catalog,table_schema,table_name,table_type
public,system,chunks,BASE TABLE
public,system,columns,BASE TABLE
public,system,operations,BASE TABLE
public,information_schema,tables,VIEW
public,information_schema,columns,VIEW

Query execution complete in 46.036373ms

foobar_weather> set format pretty;         <-------------------- Change format at runtime to pretty 
set format pretty;
Set output format format to pretty
foobar_weather> show tables;    <-------------------- Now output of  show tables is in pretty format
show tables;
+---------------+--------------------+------------+------------+
| table_catalog | table_schema       | table_name | table_type |
+---------------+--------------------+------------+------------+
| public        | system             | chunks     | BASE TABLE |
| public        | system             | columns    | BASE TABLE |
| public        | system             | operations | BASE TABLE |
| public        | information_schema | tables     | VIEW       |
| public        | information_schema | columns    | VIEW       |
+---------------+--------------------+------------+------------+

Query execution complete in 1.382642637s
```

**Review by crepererum — APPROVED** [review]

File: src/commands/sql/repl_command.rs [hunk 1]
```diff
@@ -71,6 +72,10 @@ impl TryInto<ReplCommand> for String {
             })
         } else if commands.len() == 2 && commands[0] == "show" && commands[1] == "databases" {
             Ok(ReplCommand::ShowDatabases)
+        } else if commands.len() == 3 && commands[0] == "set" && commands[1] == "format" {
```
Comment (domodwyer) [inline src/commands/sql/repl_command.rs:75]: This if might be cleaner if expressed using [slice patterns](https://doc.rust-lang.org/edition-guide/rust-2018/slice-patterns.html) - it's growing quickly! Something like:

```rust
        match commands.as_slice() {
            ["help"] => todo!(),
            ["observer"] => todo!(),
            ["exit"] => todo!(),
            ["quit"] => todo!(),
            ["use", "database"] => todo!(), // Error; no DB name specified
            ["use", "database", name] => todo!(),
            ["set", "format", fmt] => todo!(),
            // etc...
        };
```

**Review by domodwyer — APPROVED** [review]
Love how little code had to change 🎉

**Comment by alamb** [comment]
> Also: it would probably be helpful to add some kind of start and end markers before and after the CSV if it is used during some interactive session (like --- CSV START ---/--- CSV END ---).

@crepererum I have filed https://github.com/influxdata/influxdb_iox/issues/1411 for this improvement. I think it would help if there was some information about the usecase if you have any (aka *why* this would be helpful).

File: src/commands/sql/repl_command.rs (hunk 1 above)
Comment (alamb) [reply src/commands/sql/repl_command.rs:75]: Great idea -- in https://github.com/influxdata/influxdb_iox/pull/1412 -- it looks much better

**Pull request merged**